internal/annotation/annotations_test.go

```diff
@@ -455,15 +454,127 @@ func TestAnnotationKindValidation(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			for k, s := range tc.annotations {
 				assert.Equal(t, s.known, IsKnown(k))
-				assert.Equal(t, s.valid, ValidForKind(k8s.KindOf(tc.obj), k))
+				assert.Equal(t, s.valid, ValidForKind(kindOf(tc.obj), k))
 			}
 		})
 	}
 }
 
+func TestMatchIngressClass(t *testing.T) {
+
+	// This is a matrix test, we are testing the annotation parser
+	// across various annotations, with two options:
+	// ingress class is empty
+	// ingress class is not empty.
+	tests := map[string]struct {
+		fixture metav1.ObjectMetaAccessor
+		// these are results for empty and "contour" ingress class
+		// respectively.
+		want []bool
+	}{
+		"ingress nginx kubernetes.io/ingress.class": {
+			fixture: &v1beta1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "incorrect",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"kubernetes.io/ingress.class": "nginx",
+					},
+				},
+			},
+			want: []bool{false, false},
+		},
+		"ingress nginx contour.heptio.com/ingress.class": {
+			fixture: &v1beta1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "incorrect",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"contour.heptio.com/ingress.class": "nginx",
+					},
+				},
+			},
+			want: []bool{false, false},
+		},
+		"ingress contour kubernetes.io/ingress.class": {
+			fixture: &v1beta1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "incorrect",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"kubernetes.io/ingress.class": DEFAULT_INGRESS_CLASS,
+					},
+				},
+			},
+			want: []bool{true, true},
+		},
+		"ingress contour contour.heptio.com/ingress.class": {
+			fixture: &v1beta1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "incorrect",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"contour.heptio.com/ingress.class": DEFAULT_INGRESS_CLASS,
+					},
+				},
+			},
+			want: []bool{true, true},
+		},
+		"no annotation": {
+			fixture: &v1beta1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "noannotation",
+					Namespace: "default",
+				},
+			},
+			want: []bool{true, false},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			cases := []string{"", DEFAULT_INGRESS_CLASS}
+			for i := 0; i < len(cases); i++ {
+				got := MatchesIngressClass(tc.fixture, cases[i])
+				if tc.want[i] != got {
+					t.Errorf("IngressClass match(%v): expected %v, got %v", tc.fixture, tc.want[i], got)
```

Comment:
We should include the configured ingress class:
```suggestion
t.Errorf("matching (%v) against ingress class %q: expected %v, got %v", tc.fixture, cases[i], tc.want[i], got)
```

Reply:
Agreed, I'll do this now.